subsys/bluetooth/controller/ll_sw/nordic/lll/lll_df.c

```diff
@@ -65,44 +67,61 @@ uint8_t lll_df_ant_num_get(void)
 	return radio_df_ant_num_get();
 }
 
-/* @brief Function initializes transmission of Constant Tone Extension.
- *
- * @param[in] type      Type of CTE: AoA, AoD 1us, AoD 2us
- * @param[in] length    Duration of CTE in 8us units
- * @param[in] ant_num   Number of antennas in switch pattern
- * @param[in] ant_ids   Antenna identifiers that create switch pattern.
- *
- * @Note Pay attention that first two antenna identifiers in a switch
- * pattern has special purpose. First one is used in guard period, second
- * in reference period. Actual switching is processed from third antenna.
+#if defined(CONFIG_BT_CTLR_DF_ADV_CTE_TX)
+/* @brief Function enables transmission of Constant Tone Extension.
  *
- * In case of AoA mode ant_num and ant_ids parameters are not used.
+ * @param lll_sync        Pointer to LLL sync. object associated with
+ *                        periodic advertising event.
+ * @param pdu             Pointer to PDU that will be transmitted.
+ * @param[out] cte_len_us Pointer to store actual CTE length in [us]
  */
-void lll_df_conf_cte_tx_enable(uint8_t type, uint8_t length,
-			       uint8_t ant_num, uint8_t *ant_ids)
+void lll_df_cte_tx_enable(struct lll_adv_sync *lll_sync, const struct pdu_adv *pdu,
+			  uint32_t *cte_len_us)
 {
-	if (type == BT_HCI_LE_AOA_CTE) {
-		radio_df_mode_set_aoa();
-		radio_df_cte_tx_aoa_set(length);
-	}
-#if defined(CONFIG_BT_CTLR_DF_ANT_SWITCH_TX) || \
-	defined(CONFIG_BT_CTLR_DF_ANT_SWITCH_RX)
-	else {
-		radio_df_mode_set_aod();
+	if (pdu->adv_ext_ind.ext_hdr_len) {
+		const struct pdu_adv_ext_hdr *ext_hdr;
 
-		if (type == BT_HCI_LE_AOD_CTE_1US) {
-			radio_df_cte_tx_aod_2us_set(length);
+		ext_hdr = &pdu->adv_ext_ind.ext_hdr;
+
+		/* Check if particular extended PDU has cte_info. It is possible that
+		 * not all PDUs in a periodic advertising chain have CTE attached.
+		 * Nevertheless whole periodic advertising chain has the same CTE
+		 * configuration. Checking for existence of CTE configuration in lll_sync
+		 * is not enough.
+		 */
+		if (ext_hdr->cte_info) {
+			const struct lll_df_adv_cfg *df_cfg;
+
+			df_cfg = lll_adv_sync_extra_data_peek(lll_sync);
+			LL_ASSERT(df_cfg);
+
+			df_cte_tx_configure(df_cfg->cte_type, df_cfg->cte_length,
+					    df_cfg->ant_sw_len, df_cfg->ant_ids);
```

Comment:
You use enough fields of the `df_cfg` struct that it might make sense to actually use `const struct lll_df_adv_cfg *df_cfg` as the argument to the function, unless there are scenarios where we need to call this function without the access to such a struct?

Reply:
Changed to struct instead of list of parameters.
Interesting thing, I did some measurements. Flash usage after the change decreased by 16B while number of instructions related with the `df_cte_tx_configure` was larger bu 3 instructions (hope I didn't missed any instruction while counting).

Reply:
That's curious; unless you updated the function besides the parameters, it should have resulted to the same.

Reply:
Body of the function `df_cte_tx_configure` has changed due to changed arguments list. 
Take also into consideration that compiler may inline the body of the function. That may save additional bytes.